src/components/EntityPage/Participant/Utils/Holder.js

```diff
@@ -1,99 +1,108 @@
-      this.children = props.children;
-    }
-
-    this.state = {activeTab: this.tabIDs[0]};
+    this.state = {hovered: false};
   }
 
-  /**
-   * Makes the buttons for the tabs.
-   *
-   * @param holder This, as it is needed inside a nested class
-   * @returns {*|Array|*[]}
-   */
-  makeTabButtons(holder) {
-    return this.tabIDs.map(tabId => {
+  render() {
 
-      const isActive = tabId === this.state.activeTab;
+    const tabId = this.props.tabId;
+    const isActive = this.props.isActive;
 
-      //needs to be a class to use setState to do the hovering. Would do in CSS, but with emotion it's too complicated and hacky
-      class Button extends React.Component {
-        constructor(props) {
-          super(props);
+    //uses a fake Link: we want the same style as a SecondaryNavMenu, and this is the key to it
+    return (
+      <li
 
-          this.state = {hovered: false};
+        style={
+          {
+            borderRight: (isActive || this.state.hovered) ? "5px solid #e83a9c" : "5px solid transparent"
+          }
         }
 
-        render() {
-          //uses a fake Link: we want the same style as a SecondaryNavMenu, and this is the key to it
-          return (
-            <li
-              style={
-                {
-                  borderRight: (isActive || this.state.hovered) ? "5px solid #e83a9c" : "5px solid transparent"
-                }
-              }
-
-              onClick={ () => holder.setState({activeTab: tabId})}
-
-              /*
-              Would normally use a function to do the toggling, but JS classes are dumb and their "this" is not actually
-              their "this" when used inside a call
-               */
-              onMouseEnter={ () => this.setState({hovered: true}) }
-
-              onMouseLeave={ () => this.setState({hovered: false}) }
-            >
-              <Link
-
-                style={{borderBottom: "none", margin: "0 auto"}}
-
-                to={'#summary'}
-
-                className={(isActive ? 'active' : '')}>
-                {tabId}
-              </Link>
-            </li>
-          )
-        }
-      }
+        onClick={ () => this.props.clickEvent(tabId) }
+
+        /*
+        Would normally use a function to do the toggling, but JS classes are dumb and their "this" is not actually
+        their "this" when used inside a call
+         */
+        onMouseEnter={ () => this.setState({hovered: true}) }
+
+        onMouseLeave={ () => this.setState({hovered: false}) }
+      >
+        <Link
+
+          style={{borderBottom: "none", margin: "0 auto"}}
+
+          to={'#summary'}
+
+          className={(isActive ? 'active' : '')}>
+          {tabId}
+        </Link>
+      </li>
+    )
+  }
+}
+
+class Holder extends React.Component {
+
+  constructor(props) {
+    super(props);
 
-      return <Button />
-    })
+    this.state = {activeTab: (Array.isArray(props.children) ? props.children[0].props.label : props.children.props.label)};
```

Comment:
Leave it like that now that it works, but I think it would have been better to keep an index here, to avoid having to read the props of children.